enhancements/installer/cloud-custom-dns.md

```diff
@@ -0,0 +1,658 @@
+running in the bootstrap node) configures the application LBs to enable
+in-cluster and external accesss to cluster services. The Ingress Controllers
+will continue to update the DNSRecord ConfigResource(CR) with the [information]
+(https://github.com/openshift/api/blob/master/operatoringress/v1/types.go#L39)
+required to configure DNS for these services.
+
+9. The MCO will read `DNSName` from the DNSRecord CR. The `DNSName` corresponds
+to the hostname of the Ingress LB that would be added to the DNS record. MCO
+resolves this `DNSName` to the internal igress LB's IP and pass it as an input
+parameter to runtimefg which is started as an initContainer in the CoreDNS pod
+mentioned in step 5.
+
+10. With the 2 parameters to `runtimecfg` mentioned in steps 5 and 8, it can
+update the CoreFile for the CoreDNS pod with DNS entries for api-int and
+ingress thus providing in-cluster DNS resolution required for the bootstrap
+phase. 
+
+11. In addition, `runtimecfg` also updates the platform specific Status field
+within the `Infrastructure` CR with the api and api-int LN DNS names obtained
+from the `lbConfigforDNS` ConfigMap. It also updates the same Status field
+with the hostname or IP address (depending on the platform) of the internal and
+external LBs for the *.apps service. This is necessary because the user can
+refer to these fields to configure their custom DNS.
+
+12. The machine-config-operator on the master nodes, starts the CoreDNS pod if
+the Spec field in the Infrastructure CR indicates that the custom DNS feature
+is enabled. As in the case of the CoreDNS pod that was run on the bootstrap node,
+the CoreDNS pod running on the master node also runs `runtimecfg` as an
+initContainer. `runtimecfg` will be passed in the IP addresses for the internal
+and external api servers and also for the ingress service. `runtimecfg` uses
+this information to generate the CoreDNS CoreFile. 
+
+### API Extensions
+
+1. A new ConfigMap called `lbConfigforDNS` is created by the Installer. It can
+be created in any namespace but we are choosing to create it in the same
+namespace as the CoreDNS pods. Hence, the namespace name would be constructed
+as: openshift-$platform_name-infra
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: LBConfigforDNS
+  namespace: openshift-aws-infra
+data:
+  internal-api-lb-dns-name: "abc-123"
+  external-api-lb-dns-name: "xyz-456"
+```
+
+1. The AWSPlatformStatus within the PlatformStatus field of the Infrastructure
+ConfigResource (CR) is updated to contain all the DNS config required for the
+in-cluster CoreDNS solution. This same CR is available to the user post a
+successful cluster install, to configure their own DNS solution.
+
+```go
+type AWSPlatformStatus struct {
+        <snip>
+        // AWSClusterDNSConfig contains all the DNS config required to configure a custom DNS solution.
+        // +optional
+        AWSClusterDNSConfig *ClusterDNSConfig `json:"awsClusterDNSConfig,omitempty"`
```

Comment:
Is it better to add this to the `PlatformStatus` parent struct? I believe that this was discussed before but can we revisit the subject? It seems like it could be added there as the same line seems like one option for adding to AWS, GCP, Azure ... could be something like 
```
// ClusterDNSConfig contains all the DNS configuration required to configure a custom DNS solution.
// +optional
ClusterDNSConfig `json:",inline"`
```